[36기 임채동] Main페이지 Release 컴포넌트, BE Data 협의 및 반영 완료

Chaedie opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)

- [x] 기능 추가
- [x] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)

- 메인 페이지 / Release 컴포넌트 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

- 신 상품들을 보여주는 네 개의 카드를 전시
- 마우스를 올리면 컬러, 사이즈를 선택하고 장바구니에 담는 흰색 카드로 변경

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)

- ### CSS를 통해 마우스 Hover시 배경이 생기는 구현을 처음 배움

  > - CSS :hover를 이용해 마우스 온 시 흰색 카드가 나타나도록 하였습니다.이 때 트랜지션을 줘서 부드럽게 만들어지도록 하였습니다. 또한 흰색 카드에는 색깔 버튼과 사이즈 클릭 버튼 등의 버튼이 있었습니다. 처음 UI를 구성했을 땐 딱딱하게 끊어지는 모션을 보여주는 등 UI구성에 애를 먹었지만, 여러 방법을 익히며 자연스러운 형태로 만들어 냈습니다.

<br />

- ### 구현이 다 끝났다고 생각했지만 디테일하게 추가할 부분이 많았습니다.

  > - 레이아웃을 완성하고, 1차로 코드 리뷰를 받았습니다. 리뷰를 다 반영하고 새로이 확인하는 중에 추가로 구현해야겠다는 부분들이 많이 보였습니다. 이전에 완성했다고 생각하고 리뷰를 요청했지만, 그 요청이 무색하게 컬러 선택 버튼 클릭 이벤트, 사이즈 버튼 클릭 이벤트, 장바구니 버튼 활성화 로직 등을 구현해야 했습니다.

  > - 추가 기능을 모두 구현했지만, 애초에 코드 리뷰를 요청 할 때 Main에 Merge가 가능할 정도로 완벽하게 먼저 구현을 하고 리뷰를 신청했어야 하는데, 꼼꼼하게 확인 하지 않았음을 꺠달았습니다. 앞으론 꼼꼼하게 기능을 충실히 구현하고 PR을 작성해야겠습니다.

<br />

## :: 기타 질문 및 특이 사항

Chaedie committed:
Add: Main-Release 컴포넌트 레이아웃 완성

UI 레이아웃, Hover시  Layout 완성

Chaedie committed:
Fix: Mock Data 연결

Mock Data에서 Color, Size 받아 UI생성하도록 수정

Chaedie committed:
Refactor: CSS정리

sodalite1204 commented on src/components/Product/Product.js:
```diff
@@ -0,0 +1,27 @@
+      <div className="tagContainer">
+        <div className={`tag ${tag}`}>{tag}</div>
+      </div>
+      <p className="desc">{desc}</p>
```
변수명 지을 때 desc 와 같은 축약어는 최대한 피해주시는 게 좋습니다.
description일까요? 어떤 태그인지 사람에 따라서는 이해할 수 없을 것 같습니다.

sodalite1204 commented on src/components/Product/Product.scss:
```diff
@@ -0,0 +1,99 @@
+        border-radius: 1rem;
+      }
+    }
+    .cartBtn {
```
가독성을 위해서 선택자간 한 줄 띄어주세요!

sodalite1204 commented on src/components/Product/Product.scss:
```diff
@@ -0,0 +1,99 @@
+      display: block;
+      margin-top: 1.5rem;
+      padding: 0.5rem 2rem;
+      background-color: grey;
+      color: white;
```
color 값들은 border 밑에 작성해주는 게 컨벤션입니다.
대체적으로 잘 작성해주셨지만, 꼼꼼하게 다른 부분도 살펴보는 습관이 있으면 더 좋을 것 같아요!

sodalite1204 commented on src/pages/Main/Release/Release.js:
```diff
@@ -0,0 +1,74 @@
+        </button>
+      </div>
+      <div className="carouselContainer">
+        {products.map(product => (
+          <Product
+            key={product.id}
+            tag={product.tag}
+            desc={product.desc}
+            price={product.price}
+            imgUrl={product.imgUrl}
+            colorList={product.colorList}
+            sizeList={product.sizeList}
+          />
+        ))}
```
```suggestion
{products.map(product => {
          const { id, tag, desc, price, imgUrl, colorList, sizeList } = product;
          return (
            <Product
              key={id}
              tag={tag}
              desc={desc}
              price={price}
              imgUrl={imgUrl}
              colorList={colorList}
              sizeList={sizeList}
            />
          );
        })}
```
return 문을 작성하면 그 위에서 구조분해 할당 할 수 있습니다!

sodalite1204 commented on src/pages/Main/Release/Release.js:
```diff
@@ -0,0 +1,74 @@
+
+export default Release;
+
+const products = [
```
지금은 변수처럼 작성을 해주셨는데 이 부분에 관해서는 백엔드에서 받아오는 데이터이지 않을까라는 생각이드네요! 이럴 때는 목데이터 활용해서 구조 잡아주시는 게 좋을 것 같아요. 그리고 이 데이터의 구조는 백엔드와 상이된 부분일까요? 백엔드 모델링에 따라서 구조가 변화하는 부분도 있기 때문에 담당 백엔드 부분이랑 소통해보세요:)

sodalite1204 reviewed (commented):
채동님 리뷰 확인해주세요:) 다른 PR에 달아드린 부분도 있어서 확인 같이 해주세요!

Chaedie committed:
WIP

Chaedie committed:
Add: Mock Data 사용 형태로 데이터 구조 변경

fetch함수 사용하여 Mock Data 불러오는 형식으로 데이터 구조 변경
Color, Size 버튼 선택 시 장바구니 담기 버튼 활성화 되도록 변경

Chaedie committed:
Refactor: Console.log 제거

Chaedie committed:
Merge Main

Chaedie committed:
Conflict 해결

Chaedie committed:
Style: CSS 변경

Card Shadow 추가
Slide State 시인성 강화 (Opacity 조절)

Chaedie committed:
Style: BestSeller Component Opacity 조절

Chaedie committed:
Refactor: BestSeller Component Fetch문 수정

Try-Catch문 활용하여 Error Handling추가
Async-Await 활용하여 가독성 개선
setTimeout에 불필요한 10ms => 0ms 로 변경

Chaedie committed:
Refactor: 불필요한 console.log 삭제

Chaedie committed:
Refactor: 컴포넌트 재사용화

Components/SlideState 공용 컴포넌트화
=> CategorySlideState 삭제, BestSeller 내에서 Component 재사용

Chaedie committed:
Refactor: Data Fetch 실행 시점 변경

RecommendProducts 컴포넌트 내에서 Data Fetch하는 방식에서
Main 컴포넌트에서 Data Fetch하여 props로 던져주는 방식으로 변경

Chaedie committed:
Refactor: Product 컴포넌트 onClick Event Handler 하나로 합침

Chaedie committed:
WIP

Chaedie committed:
Refactor: config.js를 활용한 URI Import

필요없는 Mock Data 삭제

Chaedie committed:
Merge branch 'main' into feature/main-release

Chaedie committed:
Refactor: release => new로 명칭 통일 (with. config.js)

sodalite1204 commented on src/components/Product/Product.js:
```diff
@@ -0,0 +1,64 @@
+      </div>
+      <div className="sizePickers">
+        {sizeList.map(({ size }) => {
+          // size = size === 'null' ? '하나의' : size;
```
채동님 이 부분 어떻게 할지 확인해서 필요없다면 삭제해서 올려주세요! 
나중에 추가해야 할 경우, 추가해주세요:)

sodalite1204 reviewed (commented):
리뷰 확인해주세요!

Chaedie committed:
Refactor: 불필요한 주석 삭제 (리뷰 반영)

sodalite1204 reviewed (commented):
LGTM 👍
